extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/indexing/MSQCompactionRunner.java

```diff
@@ -159,6 +161,41 @@ public TaskStatus runCompactionTasks(
       TaskToolbox taskToolbox
   ) throws Exception
   {
+    for (Map.Entry<Interval, DataSchema> intervalDataSchema : intervalDataSchemas.entrySet()) {
+      if (Boolean.valueOf(true).equals(intervalDataSchema.getValue().getHasRolledUpSegments())) {
+        for (AggregatorFactory aggregatorFactory : intervalDataSchema.getValue().getAggregators()) {
+          // Don't proceed if either:
+          // - aggregator factory differs from its combining factory
+          // - input col name is different from the output name (idempotent)
```

Comment:
If we still have this check, why have we removed it from `ClientCompactionRunnerInfo`?

Reply:
This check is now conditional on the presence of rolled-up segments.